apps/frontend/src/pages/access-management/access-management.tsx

```diff
@@ -142,13 +149,57 @@ const AccessManagement = (): JSX.Element => {
 		[onDeleteModalOpen],
 	);
 
+	const setValidPage = useCallback(() => {
```

Comment:
instead of this helper let's add useEffect on totalItemsCount change in usePagination, and update onPageChange inside it to handle this boundary

```tsx
    useEffect(() => {
         onPageChange(page)
    }, [totalItemsCount])
```

Reply:
I added it to `use-pagination.hook.ts`, and removed `setValidPage`
```
	const onPageChange = useCallback(
		(newPage: number) => {
			const totalPages = calculateTotalPages(pageSize, totalItemsCount);
			const validPage = Math.min(newPage, totalPages);

			setPage(validPage);
		},
		[page, pageSize, totalItemsCount],
	);

	useEffect(() => {
		onPageChange(page);
	}, [totalItemsCount]);
```
I hope I understood correctly. It seems to work

Reply:
yes, looks good to me